packages/alfa-rules/test/common/expectation/get-colors.spec.tsx

```diff
@@ -130,3 +131,67 @@ test("getForeground() handles mix of opacity and transparency", (t) => {
     }
   );
 });
+
+test("getForeground() handles hover context", (t) => {
+  const target = <a href="#">Link</a>;
+
+  h.document(
+    [<html>{target}</html>],
+    [
+      h.sheet([
+        h.rule.style("a:hover", {
+          color: "red",
+        }),
+      ]),
+    ]
+  );
+
+  t.deepEqual(
+    Iterable.first(getForeground(target, device, Context.hover(target)).get())
+      .get()
+      .toJSON(),
```

Comment:
To make the test more complete, we should also check that `getForeground` without context return the non-hovered colour.